### ISSUE TITLE
Bliss should be interruptible

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#1171]
Bliss can take very long to run on some difficult graphs (see #1170).

It should be interruptible.

Bliss is written in C++, so it might be possible to just throw an exception to break out of it.

**Comment by ntamas** [comment]
Okay, so just to elaborate on the idea: we should insert a call to `IGRAPH_ALLOW_INTERRUPTION()` in the main search loop of BLISS (in `bliss/graph.cc`, in the `search()` method), and if the interrupt handler function indicates that the search was interrupted, we should throw a C++ exception that allows us to break out of the search tree. We then need to catch this exception once again in igraph's own code where it invoked BLISS functions, and turn it back into a return value of `IGRAPH_INTERRUPTED`. Sounds good, but we don't have a variant of `IGRAPH_ALLOW_INTERRUPTION()` that could work in a C++ context yet - the current implementation assumes that the function it is called from will return an igraph error code and not `void`. So we need to do something like this:

```
#define IGRAPH_THROW_IF_INTERRUPTED(exc) \
        do { \
            if (igraph_i_interruption_handler && igraph_i_interruption_handler(NULL) != IGRAPH_SUCCESS) throw exc; \
        } while (0)
```

and then, in the BLISS C++ code, we need to call:

```
IGRAPH_THROW_IF_INTERRUPTED(some_bliss_specific_exception)
```

**Comment by szhorvat** [comment]
We should contact Bliss's authors before going ahead, so we can decide on the best place to add the interruption check. I can do this.

**Comment by szhorvat** [comment]
Just a note for others not to work on this. Tommi Junttila kindly provided a prerelease with interruption functionality. Once tested, I'll send a PR.

**Comment by szhorvat** [comment]
Fixed by #1615.
